main.cpp

```diff
@@ -92,6 +92,59 @@ good to go!
  */
 
 #include <iostream>
+
+struct FloatType
+{
+public:
+    float add(float lhs, float rhs) { return lhs + rhs; }
+    float subtract(float lhs, float rhs) { return lhs - rhs; }
+    float multiply(float lhs, float rhs) { return lhs * rhs; }
+    float divide(float lhs, float rhs)
+    {
+        if (rhs == 0.0f)
+        {
+            std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        return lhs / rhs; 
+    }
+};
+
+struct DoubleType
+{
+public:
+    double add(double lhs, double rhs) { return lhs + rhs; }
+    double subtract(double lhs, double rhs) { return lhs - rhs; }
+    double multiply(double lhs, double rhs) { return lhs * rhs; }
+    double divide(double lhs, double rhs)
+    {
+        if (rhs == 0.0)
+        {
+            std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        return lhs / rhs; 
+    }
+};
+
+struct IntType
+{
+public:
+    int add(int lhs, int rhs) { return lhs + rhs; }
+    int subtract(int lhs, int rhs) { return lhs - rhs; }
+    int multiply(int lhs, int rhs) { return lhs * rhs; }
+    int divide(int lhs, int rhs)
+    {
+        if (rhs == 0)
+        {
+            std::cout << "error, integer division by zero will crash the program!\nreturn lhs" << std::endl;
+            return lhs;
+        }
```

Comment:
This `std::cout` statement does not produce an identical message when compared to the expected project output using the diffchecker linked in the instructions.
- [ ] revise

Reply:
I used the diffchecker, it showed it as identical. I've refactored it to two lines to make it more clear that it produces two lines of output though.